webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/apidata/JsonResponseObjects.scala

```diff
@@ -103,6 +135,581 @@ object RenderInheritedProperties {
+          case nodes.MachineInfo(id, VirtualMachineType(vm), serial, manufacturer) =>
+            MachineInfo(Some(id), MachineType.VirtualMachineType, Some(vm), manufacturer, serial)
+        }.getOrElse {
+          MachineInfo(None, MachineType.NoMachine, None, None, None)
+        }
+      }
+    }
+
+    /**
+     * The structure does not directly match the domain object, nodeKind is a contextual value of the node added to each AgentInfo
+     */
+    final case class Management(
+        name:         AgentType,
+        version:      Option[AgentVersion],
+        capabilities: Chunk[String],
+        nodeKind:     NodeKind
+    )
+    object Management {
+      implicit val transformer: Transformer[NodeInfo, Chunk[Management]] = (info: NodeInfo) => {
+        val agents = info.agentsName.map { agent =>
+          val capabilities = agent.capabilities.map(_.value).toList.sorted
+          Management(
+            agent.agentType,
+            agent.version,
+            Chunk.fromIterable(capabilities),
+            info.nodeKind
+          )
+        }
+        Chunk.fromIterable(agents)
+      }
+    }
+
+    /**
+     * The structure does not directly match the domain object, it aggregates some fields
+     */
+    final case class ManagementDetails(
+        cfengineKeys: Chunk[SecurityToken],
+        cfengineUser: String
+    )
+    object ManagementDetails {
+      implicit val transformer: Transformer[NodeInventory, ManagementDetails] = Transformer
+        .define[NodeInventory, ManagementDetails]
+        .withFieldComputed(_.cfengineKeys, n => Chunk.fromIterable(n.agents.map(_.securityToken)))
+        .withFieldComputed(_.cfengineUser, _.main.rootUser)
+        .buildTransformer
+    }
+
+    final case class DisplayDateTime(value: DateTime) extends AnyVal
+
+    /**
+     * Needed because datetime is serialized differently and therefore needs the DisplayDateTime wrapper
+     */
+    final case class Bios(
+        name:        String,
+        editor:      Option[SoftwareEditor],
+        version:     Option[Version],
+        quantity:    Int,
+        description: Option[String],
+        releaseDate: Option[DisplayDateTime]
+    )
+    object Bios {
+      implicit val transformer: Transformer[domain.Bios, Bios] = Transformer.derive[domain.Bios, Bios]
+    }
+
+    /**
+     * Needed because fields are renamed from the domain object
+     */
+    final case class Controller(
+        name:                     String,
+        @jsonField("type") cType: Option[String],
+        quantity:                 Int,
+        description:              Option[String],
+        manufacturer:             Option[Manufacturer]
+    )
+    object Controller {
+      implicit val transformer: Transformer[domain.Controller, Controller] = Transformer.derive[domain.Controller, Controller]
+    }
+
+    /**
+     * Needed for "swap", which has no mountpoint but the field is not optional in the domain object
```

Comment:
It's ok to use `none` for swap mountpoint, it's what is done in fstab, so we're not going out of standard path here

Reply:
I added it to the the current API https://github.com/Normation/rudder/pull/5656/commits/b591687a7572efef195c57d3d289553551157d4c